include/boost/uuid/uuid_io.hpp

```diff
@@ -146,23 +146,32 @@ inline wchar_t to_wchar(size_t i) {
 
 } // namespace detail
 
-inline std::string to_string(uuid const& u)
+template<class OutputIterator>
+bool to_chars(uuid const& u, OutputIterator out, size_t buffer_size)
 {
-    std::string result;
-    result.reserve(36);
-
+    if (buffer_size < 36) {
+        return false;
+    }
     std::size_t i=0;
     for (uuid::const_iterator it_data = u.begin(); it_data!=u.end(); ++it_data, ++i) {
         const size_t hi = ((*it_data) >> 4) & 0x0F;
-        result += detail::to_char(hi);
+        *out++ = detail::to_char(hi);
 
         const size_t lo = (*it_data) & 0x0F;
-        result += detail::to_char(lo);
+        *out++ = detail::to_char(lo);
 
         if (i == 3 || i == 5 || i == 7 || i == 9) {
-            result += '-';
+            *out++ = '-';
         }
     }
+    return true;
+}
+
+inline std::string to_string(uuid const& u)
+{
+    std::string result;
+    result.reserve(36);
+    to_chars(u, std::back_inserter(result), result.capacity());
```

Comment:
This should probably use `resize` and pass a pointer to the string `data()`.

Reply:
Will do.